src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/storage/OcrValidationRetryManagerTest.java

```diff
@@ -0,0 +1,144 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services.storage;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.testcontainers.containers.DockerComposeContainer;
+import uk.gov.hmcts.reform.bulkscanprocessor.config.IntegrationTest;
+import uk.gov.hmcts.reform.bulkscanprocessor.util.TestStorageHelper;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.bulkscanprocessor.helper.DirectoryZipper.zipDir;
+
+@IntegrationTest
+class OcrValidationRetryManagerTest {
+    private static final String CONTAINER_NAME = "bulkscan";
+
+    private static final String SAMPLE_ZIP_FILE_NAME = "1_24-06-2018-00-00-00.zip";
+
+    @Autowired
+    private OcrValidationRetryManager ocrValidationRetryManager;
+
+    @Autowired
+    private LeaseAcquirer leaseAcquirer;
+
+    private static DockerComposeContainer dockerComposeContainer;
+    private static String dockerHost;
+
+    private BlobContainerClient testContainer;
+
+    @BeforeEach
+    void setUp() {
+
+        BlobServiceClient blobServiceClient = new BlobServiceClientBuilder()
+            .connectionString(String.format(TestStorageHelper.STORAGE_CONN_STRING, dockerHost, 10000))
+            .buildClient();
+
+        testContainer = blobServiceClient.getBlobContainerClient(CONTAINER_NAME);
+        if (!testContainer.exists()) {
+            testContainer.create();
+        }
+    }
+
+    @AfterEach
+    void cleanUp() {
+        if (testContainer.exists()) {
+            testContainer.delete();
+        }
+    }
+
+    @BeforeAll
+    static void initialize() {
+        File dockerComposeFile = new File("src/integrationTest/resources/docker-compose.yml");
+
+        dockerComposeContainer = new DockerComposeContainer(dockerComposeFile)
+            .withExposedService("azure-storage", 10000)
+            .withLocalCompose(true);
+
+        dockerComposeContainer.start();
+        dockerHost = dockerComposeContainer.getServiceHost("azure-storage", 10000);
+    }
+
+    @AfterAll
+    static void tearDownContainer() {
+        dockerComposeContainer.stop();
+    }
+
+    @Test
+    void should_read_blob_and_save_metadata_in_database_when_zip_contains_metadata_and_pdfs()
+        throws Exception {
+        //Given
+        uploadToBlobStorage(SAMPLE_ZIP_FILE_NAME, zipDir("zipcontents/ok"));
+
+        BlobClient blobClient = testContainer.getBlobClient(SAMPLE_ZIP_FILE_NAME);
+
+        leaseAcquirer.ifAcquiredOrElse(
```

Comment:
this needs to be changed?
now you do not have anything with leaseAcquirer.
you can upload and get lease id. there should be sth `uploadToBlobStorageAndLease` method